### PR TITLE
Make OpenCV modules available as crate features.

### PR DESCRIPTION
**Pull request by nsabovic**
Use cargo features to select modules.

This is addressing https://github.com/twistedfall/opencv-rust/issues/297. The biggest issue is not breaking existing builds, so we have to respect `OPENCV_MODULES_(WHITELIST|BLACKLIST)`.  Since existing packages do not have features defined this has to keep functioning even with no features. There are two options here, either make it function like a whitelist where no whitelist means keep everything or declare `default` feature with everything included. The first option wouldn't work because features have to be additive. Otherwise adding a dependency with feature `core` would all of the sudden make a dependency with no features stop compiling.

A module that is required but not found will not be compiled. This allows default features to have every possible module and not worry about what's installed on the machine.

### CODE REVIEW

**Comment by twistedfall** [comment]
I would say `OPENCV_MODULES_(WHITELIST|BLACKLIST)` should go, there is little sense in having module features and those env vars simultaneously, they serve the same purpose. Can you please also not include the `core` module as a feature. The crate isn't really functional without it.

**Comment by twistedfall** [comment]
Also the list is missing some modules, here is the full list of modules (which I test locally):
	alphamat
	aruco
	barcode
	bgsegm
	bioinspired
	calib3d
	ccalib
	core
	cudaarithm
	cudabgsegm
	cudacodec
	cudafeatures2d
	cudafilters
	cudaimgproc
	cudaobjdetect
	cudaoptflow
	cudastereo
	cudawarping
	cvv
	dnn
	dnn_superres
	dpm
	face
	features2d
	flann
	freetype
	fuzzy
	gapi
	hdf
	hfs
	highgui
	img_hash
	imgcodecs
	imgproc
	intensity_transform
	line_descriptor
	mcc
	ml
	objdetect
	optflow
	ovis
	phase_unwrapping
	photo
	plot
	quality
	rapid
	rgbd
	saliency
	sfm
	shape
	stereo
	stitching
	structured_light
	superres
	surface_matching
	text
	tracking
	video
	videoio
	videostab
	viz
	xfeatures2d
	ximgproc
	xobjdetect
	xphoto
	wechat_qrcode

File: build.rs [hunk 1]
```diff
@@ -154,6 +155,9 @@ fn make_modules(opencv_dir: &Path) -> Result<()> {
 			.map(|e| e.trim())
 			.collect::<HashSet<_>>()
 		);
+	let feature_whitelist: HashSet<String> = HashSet::from_iter(env::vars_os().filter_map(|(k, _)|
```
Comment (twistedfall) [inline build.rs]: I'd say it's more conventional to call `.collect()` instead of `from_iter()`. Saves you an import and keeps a more linear call chain.

**Comment by nsabovic** [comment]
Done. This will clearly break anyone currently using env vars in their builds. But it's a small break 😀...

**Pull request merged**